lib/mongoid/criteria/queryable/selector.rb

```diff
@@ -22,9 +22,9 @@ def merge!(other)
             if value.is_a?(Hash) && self[key.to_s].is_a?(Hash)
               value = self[key.to_s].merge(value) do |_key, old_val, new_val|
                 case _key
```

Comment:
its better to do `_key.to_s` here. You regex would technically match `"I'm a $ninja"` as a key.

Reply:
Thanks a million, you're right.
Done now